libvirt.go

```diff
@@ -235,9 +237,32 @@ func (d *Driver) PreCreateCheck() error {
 }
 
 func (d *Driver) Create() error {
-	b2dutils := mcnutils.NewB2dUtils(d.StorePath, "crc")
-	if err := b2dutils.CopyDiskToMachineDir(d.DiskPathURL, d.MachineName); err != nil {
-		return err
+	u, err := url.Parse(d.DiskPathURL)
+	if err != nil {
+		return fmt.Errorf("Invalid URL: %s", d.DiskPathURL)
+	}
+
+	if u.Scheme == "file" || u.Scheme == "" {
+		// For local image on Linux, we can avoid copying by making use of base image through qemu-img
+		// Local disk image so we can avoid copying by making use of backing image feature of qcow2
+		diskPath := u.Path
+		machineIsoPath := filepath.Join(d.StorePath, "machines", "crc", "crc")
+
+		log.Infof("Create disk image at %s (backing store: %s)", machineIsoPath, diskPath)
+		command := exec.Command("qemu-img", "create", "-b", diskPath, "-f", "qcow2", machineIsoPath)
```

Comment:
Instead using `qemu-img` better to check if `libvirt-go` provide some api around it so that end user doesn't need to install this command.

Reply:
agree. rather not invoke external commands when possible.

Reply:
We're not using libvirt for the disk image management. We should I think but currently we don't. We'll need/want to do that before being able to use libvirt API. I think we should handle it like this:

1. I modify this patch so that it reverts back to copying if `qemu-img` failed for any reason (including qemu-img) not found. That way if you've qemu-img, you get faster `crc` and not otherwise.

2. Port the code to make use of libvirt volume API for disk management and as part of that, `qemu-img` use will be removed.

How about that?

Reply:
sounds good to me

Reply:
Yes, that sound reasonable 👍

Reply:
1. done in the updating patch.
2. https://github.com/code-ready/machine-driver-libvirt/issues/39